faunadb/client.go

```diff
@@ -215,33 +263,49 @@ func (client *FaunaClient) newClient(basicAuth string, observer ObserverCallback
 	return &FaunaClient{
 		basicAuth:        basicAuth,
 		endpoint:         client.endpoint,
+		headers:          client.headers,
 		http:             client.http,
 		isTxnTimeEnabled: client.isTxnTimeEnabled,
+		queryTimeoutMs:   client.queryTimeoutMs,
 		lastTxnTime:      client.lastTxnTime,
 		observer:         observer,
 	}
 }
 
-func (client *FaunaClient) performRequest(expr Expr) (response *http.Response, err error) {
-	var request *http.Request
+func (client *FaunaClient) performRequest(expr Expr, configs []QueryConfig) (response *http.Response, err error) {
 
-	if request, err = client.prepareRequest(expr); err == nil {
+	if request, err := client.prepareRequest(expr, configs); err == nil {
 		response, err = client.http.Do(request)
 	}
 
 	return
 }
 
-func (client *FaunaClient) prepareRequest(expr Expr) (request *http.Request, err error) {
+func (client *FaunaClient) prepareRequest(expr Expr, configs []QueryConfig) (request *http.Request, err error) {
 	var body []byte
 
 	if body, err = json.Marshal(expr); err == nil {
-		if request, err = http.NewRequest("POST", client.endpoint, bytes.NewReader(body)); err == nil {
-			request.Header.Add("Authorization", client.basicAuth)
-			request.Header.Add("Content-Type", "application/json; charset=utf-8")
-			request.Header.Add("X-FaunaDB-API-Version", apiVersion)
-			request.Header.Add("X-Fauna-Driver", headerFaunaDriver)
+		if request, err := http.NewRequest("POST", client.endpoint, bytes.NewReader(body)); err == nil {
+			headers := map[string]string{}
```

Comment:
This seems only used to set the authorization token, which could go strait into the request headers. Dropping this variable will simplify the code.